### PR TITLE
Add database.

### PR DESCRIPTION
**Pull request by j0sh**
For now, initializes a simple key-value schema that is used for the following:

* Listens for a new block, and writes it disk. (Needed for job validity checking) https://github.com/livepeer/go-livepeer/commit/a4afb564428c3292bf461ef333e6425409f68312
* Basic versioning checks. https://github.com/livepeer/go-livepeer/commit/c5ee8b423a0dc3b4962ae14312c0867f4b4e0f3f

### CODE REVIEW

File: eth/eventservices/jobservice.go [hunk 1]
```diff
@@ -80,6 +80,11 @@ func (s *JobService) Start(ctx context.Context) error {
 		return err
 	}
 
+	s.eventMonitor.SubscribeNewBlock(context.Background(), "BlockWatcher", make(chan *types.Header), func(h *types.Header) (bool, error) {
```
Comment (ericxtang) [inline eth/eventservices/jobservice.go:83]: Just a comment - I think we should re-architect how we are subscribing to events.  There is currently a lot of duplicated subscriptions.  Every time we call `SubscribeNewBlock`, we send a new subscription request to the remote provider (could explain why our sub connection is unstable).  

Instead, we should be sending a sub request per event type, and having multiple local listeners on the same event.  

Added an issue here https://github.com/livepeer/go-livepeer/issues/411

File: common/db.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+var ErrDBTooNew = errors.New("DB Too New")
+
+var schema = `
+	CREATE TABLE IF NOT EXISTS kv (
+		key STRING PRIMARY KEY,
+		value STRING,
+		updatedAt STRING DEFAULT CURRENT_TIMESTAMP
+	);
+	INSERT OR IGNORE INTO kv(key, value) VALUES('dbVersion', '{{ . }}');
+	INSERT OR IGNORE INTO kv(key, value) VALUES('lastBlock', '0');
```
Comment (ericxtang) [inline common/db.go:32]: So for more complex structures like a job, we'll just serialize/deserialize?  What if we wanted to do a query based on an attribute?

**Review by ericxtang — APPROVED** [review]
Looks good - just a few questions and comments.  🚢

File: common/db.go (hunk 2 above)
Comment (ericxtang) [reply common/db.go:32]: nvm.  Doing review for the jobs PR.

File: common/db.go [hunk 3]
```diff
@@ -38,14 +44,34 @@ func InitDB(dbPath string) (*DB, error) {
...
+	}
+	if dbVersion > LivepeerDBVersion {
+		glog.Error("Database too new")
+		d.Close()
+		return nil, ErrDBTooNew
+	} else if dbVersion < LivepeerDBVersion {
+		// Upgrade stepwise up to the correct version using the migration
+		// procedure for each version
+	} else if dbVersion == LivepeerDBVersion {
+		// all good; nothing to do
```
Comment (j0sh) [inline common/db.go:72]: Updated this comment; hopefully it's clearer that this no-op is expected @yondonfu

**Comment by j0sh** [comment]
Updated the PR with a minor change: the DB version check/migration code was moved to immediately after schema initialization. The old code had all the prepared statements initialized *before* the version check, which would obviously fail with an outdated schema.

**Review by yondonfu — APPROVED** [review]
LGTM!

**Pull request merged**